crates/rustc_codegen_spirv/src/builder/libm_intrinsics.rs

```diff
@@ -0,0 +1,408 @@
+        "_ZN4libm4math9remainder9remainder17h07882671783553d7E" => {
+            LibmIntrinsic::Custom(LibmCustomIntrinsic::Remainder)
+        } // math::remainder::remainder
+        "_ZN4libm4math10remainderf10remainderf17hab3f628ce44e8fafE" => {
+            LibmIntrinsic::Custom(LibmCustomIntrinsic::Remainder)
+        } // math::remainderf::remainderf
+        "_ZN4libm4math6remquo6remquo17h2476dac7541ba108E" => {
+            LibmIntrinsic::Custom(LibmCustomIntrinsic::RemQuo)
+        } // math::remquo::remquo
+        "_ZN4libm4math7remquof7remquof17h8cb54bebc10b68e6E" => {
+            LibmIntrinsic::Custom(LibmCustomIntrinsic::RemQuo)
+        } // math::remquof::remquof
+        "_ZN4libm4math5round5round17hdfc544d16997c663E" => LibmIntrinsic::GLOp(GLOp::Round), // math::round::round
+        "_ZN4libm4math6roundf6roundf17hf9e34870ea5a626eE" => LibmIntrinsic::GLOp(GLOp::Round), // math::roundf::roundf
+        "_ZN4libm4math6scalbn6scalbn17hfcb5cc2162b27388E" => {
+            LibmIntrinsic::Custom(LibmCustomIntrinsic::Scalbn)
+        } // math::scalbn::scalbn
+        "_ZN4libm4math7scalbnf7scalbnf17he930032788b814a0E" => {
+            LibmIntrinsic::Custom(LibmCustomIntrinsic::Scalbn)
+        } // math::scalbnf::scalbnf
+        "_ZN4libm4math3sin3sin17h6b2fc3ac4644f1f2E" => LibmIntrinsic::GLOp(GLOp::Sin), // math::sin::sin
+        "_ZN4libm4math6sincos6sincos17h3128c207d10585c2E" => {
+            LibmIntrinsic::Custom(LibmCustomIntrinsic::SinCos)
+        } // math::sincos::sincos
+        "_ZN4libm4math7sincosf7sincosf17h481fa4ff66a116cfE" => {
+            LibmIntrinsic::Custom(LibmCustomIntrinsic::SinCos)
+        } // math::sincosf::sincosf
+        "_ZN4libm4math4sinf4sinf17h498f0ce121a8b58eE" => LibmIntrinsic::GLOp(GLOp::Sin), // math::sinf::sinf
+        "_ZN4libm4math4sinh4sinh17hdd82b50e0339f64bE" => LibmIntrinsic::GLOp(GLOp::Sinh), // math::sinh::sinh
+        "_ZN4libm4math5sinhf5sinhf17hb89e68353a4f40c1E" => LibmIntrinsic::GLOp(GLOp::Sinh), // math::sinhf::sinhf
+        "_ZN4libm4math4sqrt4sqrt17he29bb1acace0b4a0E" => LibmIntrinsic::GLOp(GLOp::Sqrt), // math::sqrt::sqrt
+        "_ZN4libm4math5sqrtf5sqrtf17h2d9d63ed22c9d3b0E" => LibmIntrinsic::GLOp(GLOp::Sqrt), // math::sqrtf::sqrtf
+        "_ZN4libm4math3tan3tan17h9459ee48dfa3f29aE" => LibmIntrinsic::GLOp(GLOp::Tan), // math::tan::tan
+        "_ZN4libm4math4tanf4tanf17h8a532d29d0ee1292E" => LibmIntrinsic::GLOp(GLOp::Tan), // math::tanf::tanf
+        "_ZN4libm4math4tanh4tanh17h18e20d55f6144b5dE" => LibmIntrinsic::GLOp(GLOp::Tanh), // math::tanh::tanh
+        "_ZN4libm4math5tanhf5tanhf17h84506970bf1bca9fE" => LibmIntrinsic::GLOp(GLOp::Tanh), // math::tanhf::tanhf
+        "_ZN4libm4math5trunc5trunc17ha75805e376ba9aa2E" => LibmIntrinsic::GLOp(GLOp::Trunc), // math::trunc::trunc
+        "_ZN4libm4math6truncf6truncf17h64b5ded9471003b1E" => LibmIntrinsic::GLOp(GLOp::Trunc), // math::truncf::truncf
```

Comment:
Looks like these are all¹ `math::$fn::$fn`, so it should be possible to grab the function name (via `tcx.item_name(def_id)`, or `tcx.def_path(def_id)` if you want the entire path at once), and check that the rest of the path conforms.

¹exceptions:
```rust
// math::erf::erfc
// math::erff::erfcf
// math::j0::y0
// math::j0f::y0f
// math::j1::y1
// math::j1f::y1f
// math::jn::yn
// math::jnf::ynf
```